aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -301,7 +301,10 @@ module aptos_framework::coin {
                 if (is_apt) {
                     object::create_sticky_object_at_address(@aptos_framework, @aptos_fungible_asset)
                 } else {
-                    object::create_sticky_object(coin_address<CoinType>())
+                    object::create_user_derived_object(
+                        @aptos_fungible_asset,
```

Comment:
why do we derive from @aptos_fungible_asset for non-apt ? that seems a bit awkward? is there nothing better to put here? even 0x1 sounds better

Reply:
derive(owner_account_address, metadata_address) = (0xA, coin_creator_address)
So this is a trick that leverages this function to generate unique address w/o a new domain separator.
I think what you mean is `derive(coin_creator_address, 0xA)`?

Reply:
actually I was using (coin_creator_address, 0x0)... so not sure what's the best tuple here. cc @davidiw

Reply:
0xA is APT FA address, and is unrelated to mapping here
`derive(coin_creator_address, 0x0) `seems better to me, or even `derive(coin_creator_address, coin_creator_address)`

Reply:
I like (coin_creator_address, coin_creator_address)... @davidiw sgty?

Reply:
No, we cannot allow this to be influenced by a third party, someone could try to create that, no? I want this to be a guaranteed safe operation.

Reply:
No, create_user_derived_object is a friend function that can only be called by 0x1.

Reply:
` public(friend) fun create_user_derived_object(creator_address: address, derive_ref: &DeriveRef)`
it is a friend function so third party cannot create that

Reply:
so this function only exists for the purpose of primary fungible store address?

Reply:
Currently this is the only use case for now. But not necessarily be for PFS address in the future.

Reply:
also the first argument to this function is `address` so the value does not matter at all with regard to security.